index.js

```diff
@@ -212,14 +212,26 @@ async function installBundler(bundlerVersionInput, platform, rubyPrefix, engine,
 }
```

Comment:
There is some logic higher in installBundler. This also needs to take care about `gems.locked` (unless `BUNDLE_GEMFILE` is set but I guess not in your case).